### PR TITLE
Enhancement/normalization dask

### LINKED ISSUE #762: normalization functions with dask
### Description of feature

Most of the normalization functions trigger the computation when the array in adata is a dask array.
This can be adjusted when using [dask-ml](https://ml.dask.org/modules/api.html#module-dask_ml.preprocessing) preprocessing functions in this instance, which are not as complete as sklearn but available for most normalization functions.

### PR DESCRIPTION
**Pull request by eroell**
**PR Checklist**

-   [x] This comment contains a description of changes (with reason)
-   [x] Referenced issue is linked: Fixes #762 
-   [x] If you've fixed a bug or added code that should be tested, add tests!
-   [x] Documentation in `docs` is updated

**Description of changes**

Allow normalization methods to work with (dense) dask array.
Suggest `ehrapy[dask]` for dependency management. Might make dask a dependency in the future.

**Technical details**
- Moved from the scikit-learn (e.g. `sklearn.preprocessing.scale`) functions to the scikit-learn classes (e.g. `sklearn.preprocessing.StandardScaler`) for preprocessing to be more synced with dask-ml (only has the classes option, e.g. `dask_ml.preprocessing.StandardScaler`). No user facing effects, updated doc.
- Added dask as optional dependency in pyproject.toml, and as mandatory dependency for test
- Added dask-ml preprocessing classes, also doc updated, which allow lazy computation of dask arrays (No MaxAbsScaler, no PowerTransformer here, Error will be raised)
- Added dask to test, following a style similar as scanpy is currently using it
- Significant performance upgrade especially in dask case by vectorizing a for loop in the scaling function
- Extensive testing, also refactored into more fine-grained tests to avoid overloading & getting hard-to-decipher error messages.

**Additional context**

Example, profiled with scalene (run below python scripts as `scalene <scriptname>.py`, demonstrating how `ep.pp.scale_norm` does not trigger the computations and is not performance bottleneck:

In memory (numpy) array
```py
import scalene
scalene.scalene_profiler.stop()
import pandas as pd
from sklearn.datasets import make_blobs as make_blobs
import ehrapy as ep
import anndata as ad
import scanpy as sc
n_individuals = 50000
n_features = 1000
n_groups = 4
chunks = 1000
data_features, data_labels = make_blobs(n_samples=n_individuals, n_features=n_features, centers=n_groups, random_state=42)
var = pd.DataFrame({"feature_type": ["numeric"] * n_features})
adata = ad.AnnData(X=data_features, obs={"label": data_labels}, var=var)
scalene.scalene_profiler.start()
ep.pp.scale_norm(adata)
ep.pp.pca(adata)
sc.pp.neighbors(adata)
ep.tl.leiden(adata)
ep.pl.pca(adata, color="leiden", save="profiling_memory_pca.png")
scalene.scalene_profiler.stop()
```
memory_profile_50000x1000

Out-of-memory (dask) array
```py
import scalene
scalene.scalene_profiler.stop()
import dask.array as da
from sklearn.datasets import make_blobs as make_blobs
import ehrapy as ep
import anndata as ad
import pandas as pd
import scanpy as sc
n_individuals = 50000
n_features = 1000
n_groups = 4
chunks = 1000
data_features, data_labels = make_blobs(n_samples=n_individuals, n_features=n_features, centers=n_groups, random_state=42)
data_features = da.from_array(data_features, chunks=chunks)
var = pd.DataFrame({"feature_type": ["numeric"] * n_features})
adata = ad.AnnData(X=data_features, obs={"label": data_labels}, var=var)
scalene.scalene_profiler.start()
ep.pp.scale_norm(adata)
ep.pp.pca(adata)
adata.obsm["X_pca"] = adata.obsm["X_pca"].compute()
sc.pp.neighbors(adata)
sc.tl.leiden(adata)
sc.pl.pca(adata, color="leiden", save="profiling_out_of_core_pca.png")
scalene.scalene_profiler.stop()
```
out_of_core_profile_50000x1000

### CODE REVIEW

File: ehrapy/anndata/_constants.py [hunk 1]
```diff
@@ -6,3 +6,13 @@
...
+
+NORM_NAMES = {
+    "StandardScaler" : "scale",
+    "MinMaxScaler" : "minmax", 
+    "MaxAbsScaler" : "maxabs",
+    "PowerTransformer" : "quantile",
+    "QuantileTransformer" : "power",
+    "RobustScaler" : "robust_scale",
+    "log_norm" : "log"
+}
```
Comment (eroell) [inline ehrapy/anndata/_constants.py:18]: these constants have been distributed across `_normalization.py` before

File: ehrapy/anndata/anndata_ext.py [hunk 2]
```diff
@@ -381,8 +381,9 @@ def set_numeric_vars(
 
     vars_idx = _get_column_indices(adata, vars)
 
-    for i in range(n_values):
-        adata.X[:, vars_idx[i]] = values[:, i]
+    adata.X[:, vars_idx] = values
```
Comment (eroell) [inline ehrapy/anndata/anndata_ext.py:384]: this is a significant speedup, 2-4x with dask on my machine

File: ehrapy/anndata/_constants.py (hunk 1 above)
Comment (Zethson) [reply ehrapy/anndata/_constants.py]: Hmm but these are not `anndata` constants, they are `normalization` constants. If we had a global constants module at the root level of ehrapy, I'd get it...

File: ehrapy/anndata/anndata_ext.py (hunk 2 above)
Comment (Zethson) [reply ehrapy/anndata/anndata_ext.py:384]: Lovely!

File: ehrapy/anndata/_constants.py (hunk 1 above)
Comment (eroell) [reply ehrapy/anndata/_constants.py]: We can also make such a global constants module at the root level of ehrapy - so far, I've taken the anndata constants to be this (since its constants "added" to anndata which in some cases can tune functionality, although for the normalization this is not yet the case)

File: ehrapy/_compat.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Since we might check whether an object is an instance of dask.array.Array
```
Comment (Zethson) [inline ehrapy/_compat.py:1]: Let's move this file to conftest.py

**Review by Zethson — APPROVED** [review]
Lovely!

File: .readthedocs.yml [hunk 4]
```diff
@@ -12,7 +12,7 @@ build:
         post_install:
             # VIRTUAL_ENV needs to be set manually for now.
             # See https://github.com/readthedocs/readthedocs.org/pull/11152/
-            - VIRTUAL_ENV=$READTHEDOCS_VIRTUALENV_PATH uv pip install .[docs]
+            - VIRTUAL_ENV=$READTHEDOCS_VIRTUALENV_PATH pip install .[docs]
```
Comment (eroell) [inline .readthedocs.yml:15]: Readthedocs fails with

```
...
running build_rust
--- stderr:
error: [Errno 2] No such file or directory: 'cargo'
---
```

for some reason. Is this ok as temporary making this build for now & optimize again later by fixing the use of uv?

File: ehrapy/anndata/_constants.py [hunk 5]
```diff
@@ -6,3 +6,13 @@
 NUMERIC_TAG = "numeric"
 CATEGORICAL_TAG = "categorical"
 DATE_TAG = "date"
+
+NORM_NAMES = {
```
[the remark this answers is not in the document]
Comment (eroell) [reply ehrapy/anndata/_constants.py]: moved back

File: ehrapy/anndata/_constants.py (hunk 1 above)
Comment (eroell) [reply ehrapy/anndata/_constants.py]: moved back now

**Comment by eroell** [comment]
weeeird, the `pull_request_target` seems to still appear, even though we fixed this to use the `pull_request` trigger in run_notebooks.yml.
But I can see the notebooks which should work triggered on `pull_request` actually pass.

Might disappear once this is merged, lets see how this behaves in future PRs.
Big thanks @flying-sheep @ilan-gold .

File: .github/workflows/run_notebooks.yml [hunk 6]
```diff
@@ -31,14 +31,8 @@ jobs:
             - name: Install UV
               run: pip install uv
```
Comment (eroell) [inline .github/workflows/run_notebooks.yml:33]: Upon suggestion of @flying-sheep
